### PR TITLE
Launchpad: Add use launchpad checklist hook to launchpad sidebar

### PR DESCRIPTION
**Pull request by jeyip**
Related to https://github.com/Automattic/wp-calypso/issues/75504

**Review time: short**
**Testing time: med**

## Proposed Changes

* Instead of using client-side definitions, fetch checklist and task definitions from the launchpad checklist API for the sidebar
* Use the fetched task definitions to render the launchpad checklist

## Testing Instructions

1. Check out this branch and run `yarn` and `yarn start` if needed. 
2. Start any new launchpad enabled site
   - Newsletter http://calypso.localhost:3000/setup/newsletter/intro 
   - Link-in-Bio http://calypso.localhost:3000/setup/link-in-bio/intro
   - VideoPress http://calypso.localhost:3000/setup/videopress/intro
   - Build http://calypso.localhost:3000/start ( select `Other` as the site goal )
   - Write http://calypso.localhost:3000/start ( select `Write & Publish` as the site goal )
3. Navigate to the Launchpad by completing onboarding
4. Verify that the number of tasks for each flow has remained the same
5. Verify that the text copy of each task for each flow has remained the same
6. Verify that the tasks for each flow behave exactly like before when clicked ( launch site button launches the site, edit site design redirects user to site editor, etc. )

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx [hunk 1]
```diff
@@ -63,43 +51,42 @@ const Sidebar = ( { sidebarDomain, siteSlug, submit, goNext, goToStep, flow }: S
...
+			launchpadChecklist,
 			siteSlug,
 			site,
 			submit,
 			globalStylesInUse && shouldLimitGlobalStyles,
 			goToStep,
 			flow,
-			isEmailVerified,
-			checklist_statuses
+			isEmailVerified
```
Comment (agrullon95) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx]: We need to update `getEnhancedTasks()` to determine the task completion statuses based on `launchpadChecklist`.  Task completion status logic was previously relying on `checklist_statuses`.

**Review by agrullon95 — CHANGES_REQUESTED** [review]

**Comment by agrullon95** [comment]
Im also noticing a delay with the sidebar data loading.


https://user-images.githubusercontent.com/10482592/234320622-1182c100-7e51-460a-b68c-6b2616a5d149.mp4

**Comment by jeyip** [comment]
> Im also noticing a delay with the sidebar data loading.

Ah interesting. It looks particularly slow here -- I imagine because you had sandboxed the public API. It wasn't typically this slow when I was testing it.

Either way, this is demonstrating an important point -- we now rely on a network request to load the sidebar task list, which may take several seconds. I know we're averse to loading states and loading spinners, but I'm unsure how else we'd convey that we're loading data.

@agrullon95 do you have thoughts?

@nuriapenya wondering if you can chime in. To be clear, loading the tasks won't normally take as long as shown in Anthony's gif.

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx (hunk 1 above)
Comment (jeyip) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx]: @agrullon95 I was planning to do this in a follow-up to keep this PR compact. How do feel about that?

Leaving it as is for the time being doesn't change the resulting behavior, and it keeps this PR compact

**Comment by sharpirate** [comment]
> I know we're averse to loading states and loading spinners, but I'm unsure how else we'd convey that we're loading data.

Can we pre-fetch the tasks as part of the loading screen that is shown right before Launchpad?

I also tried testing this without sandboxing the public API and I continued to experience long loading times

**Comment by sharpirate** [comment]
### Testing
✔️ Newsletter tasks are matching production and behave as expected
(Although there's one extra task: **Confirm email (check your inbox)**)
✔️ Link in Bio tasks are matching production and behave as expected
✔️ Free tasks are matching production and behave as expected
✔️ VideoPress tasks are matching production and behave as expected
✔️ Build tasks are matching production and behave as expected
✔️ Write tasks are matching production and behave as expected

Note: In addition to the testing instructions, I had to sync my sandbox with `jetpack/trunk` by running `bin/jetpack-downloader test jetpack-mu-wpcom-plugin trunk` since it looks like not all of our endpoint changes have been deployed yet. My understanding is we'll need to hold off on deploying this PR until then.

**Comment by jeyip** [comment]
@nuria suggested a loading screens similar to what we see in the domains onboarding page p1682518854697819/1682459093.067819-slack-CHN6J22MP

I'll get started implementing it now

**Comment by agrullon95** [comment]
I would prefer a loading placeholder (not sure what the correct name is), so users at least know content is supposed to appear in that area instead of seeing a blank space where the checklist should be.

**Comment by jeyip** [comment]
Updated the loading status @agrullon95. Some notes -- we'll probably want to modify the styling around the progress icon so that there's less content jumping after we fetch data successfully

File: packages/help-center/src/hooks/use-launchpad-checklist.tsx [hunk 2]
```diff
@@ -45,9 +45,7 @@ export const useLaunchpadChecklist = ( siteSlug: string | null, siteIntent: stri
 	const key = [ 'launchpad-checklist', siteSlug ];
 	const queryResult = useQuery( key, () => fetchLaunchpadChecklist( siteSlug, siteIntent ), {
 		retry: 3,
-		initialData: {
-			checklist: [],
-		},
+		placeholderData: { checklist: [] },
```
Comment (jeyip) [inline packages/help-center/src/hooks/use-launchpad-checklist.tsx:51]: placeholderData doesn't overwrite existing cached values ( which we _do_ store between sessions in https://github.com/Automattic/wp-calypso/blob/trunk/client/state/query-client.ts#L12-L19 )

**Comment by agrullon95** [comment]
@jeyip - I dont see the loading placeholder on the sidebar when I go through the onboarding flow and land on the launchpad.

https://user-images.githubusercontent.com/10482592/234938817-c2959e64-48c3-4d44-94d7-73a6a74bb804.mp4

**Comment by sharpirate** [comment]
### Testing

✔️ Tested a few different flows and the loading screen works as expected

**Comment by jeyip** [comment]
**Note:**

If you're testing against production, the current newsletter checklist definitions are incorrect fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Swrgcnpx%2Qzh%2Qjcpbz%2Qcyhtva%2Scebqhpgvba%2Siraqbe%2Snhgbznggvp%2Swrgcnpx%2Qzh%2Qjcpbz%2Sfep%2Ssrngherf%2Synhapucnq%2Synhapucnq.cuc%3Se%3Qq27nq3p3%2352%2Q53-og. 

Once the new `jetpack-mu-wpcom` version is released, this will automatically be resolved

**Comment by jeyip** [comment]
The bug with the newsletter checklist should be resolved @agrullon95

**Comment by sharpirate** [comment]
✔️ Just retested this against `jetpack/trunk` and it works as expected

**Review by agrullon95 — APPROVED** [review]
> The bug with the newsletter checklist should be resolved @agrullon95

I retested the Newsletter, Free, and Write flows. The tasks were marked as completed as I went through them and I'm able to confirm I no longer see the checklist bug for the newsletter flow.

**Pull request merged**